dsgrid/registry/project_registry_manager.py

```diff
@@ -355,6 +396,156 @@ def _register_dimensions_from_models(
+    ):
+        """Registers a dimension for each subset specified in the project config's subset
+        dimension groups. Appends references to those dimensions to subset_dimensions, which is
+        part of the project config.
+        """
+        with TemporaryDirectory() as tmpdir:
+            tmp_path = Path(tmpdir)
+            dimensions = []
+            subset_refs = {}
+            for subset_dimension in subset_dimensions:
+                base_dim = None
+                for ref in base_dimension_references:
+                    if ref.dimension_type == subset_dimension.dimension_type:
+                        base_dim = self._dimension_mgr.get_by_id(ref.dimension_id)
+                        break
+                assert base_dim is not None, subset_dimension
+                base_records = base_dim.get_records_dataframe()
+                for selector in subset_dimension.selectors:
+                    new_records = base_records.filter(base_records["id"].isin(selector.records))
+                    filename = tmp_path / f"{subset_dimension.name}_{selector.name}.csv"
+                    new_records.toPandas().to_csv(filename, index=False)
+                    dim = DimensionModel(
+                        filename=str(filename),
+                        name=selector.name,
+                        display_name=selector.name,
+                        dimension_type=subset_dimension.dimension_type,
+                        module=base_dim.model.module,
+                        class_name=base_dim.model.class_name,
+                        description=selector.description,
+                    )
+                    dimensions.append(dim)
+                    key = (subset_dimension.dimension_type, selector.name)
+                    if key in subset_refs:
+                        raise Exception(f"Bug: unhandled case of duplicate dimension name: {key=}")
+                    subset_refs[key] = subset_dimension
+
+            dim_model = DimensionsConfigModel(dimensions=dimensions)
+            dims_config = DimensionsConfig.load_from_model(dim_model)
+            dimension_ids = self._dimension_mgr.register_from_config(
+                dims_config, submitter, log_message, context=context
+            )
+            for dimension_id in dimension_ids:
+                dim = self._dimension_mgr.get_by_id(dimension_id)
+                key = (dim.model.dimension_type, dim.model.name)
+                subset_dim = subset_refs[key]
+                subset_dim.selector_references.append(
+                    DimensionReferenceModel(
+                        dimension_id=dimension_id,
+                        dimension_type=subset_dim.dimension_type,
+                        version="1.0.0",
+                    )
+                )
+
+    def _register_supplemental_dimensions_from_subset_dimensions(
+        self, model, context, submitter, log_message
+    ):
```

Comment:
This function should probably have a doc string similar to the one provided for the above function.